cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -546,6 +572,8 @@ int32_t cros_gralloc_driver::retain(buffer_handle_t handle)
 		.use_flags = hnd->use_flags,
 	};
 	drv = is_video_format(&descriptor) ? drv_video_ : drv_render_;
+	if (from_kms)
```

Comment:
Maybe you can check by: if ((hnd->use_flags & BO_USE_SCANOUT) && (!is_video_format(descriptor)))
and remove "from_kms"

Reply:
ok